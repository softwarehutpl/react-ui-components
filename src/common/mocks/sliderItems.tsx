import React from 'react';

const sliderItems = [{
    src: 'https://images.unsplash.com/photo-1449034446853-66c86144b0ad?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2100&q=80',
    // captureComponent: <span className='capture-class'> Image first </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1470341223622-1019832be824?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2288&q=80',
    captureComponent: <span className='capture-class'> Image second </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1448630360428-65456885c650?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2094&q=80',
    captureComponent: <span className='capture-class'> Image third </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1534161308652-fdfcf10f62c4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2174&q=80',
    captureComponent: <span className='capture-class'>Image 4th </span>,
  },  {
    src: 'https://images.unsplash.com/photo-1452827073306-6e6e661baf57?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2167&q=8',
    captureComponent: <span className='capture-class'> Image 5th</span>,
  }, {
    src: 'https://images.unsplash.com/photo-1533907650686-70576141c030?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=6000&q=80',
    captureComponent: <span className='capture-class'> some capyture </span>,
  },  {
    src: 'https://images.unsplash.com/photo-1474401915596-3c5adf84ef01?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2250&q=80',
    captureComponent: <span className='capture-class'> some capyture </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1449034446853-66c86144b0ad?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2100&q=80',
    captureComponent: <span className='capture-class'> Image first </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1470341223622-1019832be824?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2288&q=80',
    captureComponent: <span className='capture-class'> Image second </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1448630360428-65456885c650?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2094&q=80',
    captureComponent: <span className='capture-class'> Image third </span>,
  }, {
    src: 'https://images.unsplash.com/photo-1534161308652-fdfcf10f62c4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2174&q=80',
    captureComponent: <span className='capture-class'>Image 4th </span>,
  },  {
    src: 'https://images.unsplash.com/photo-1452827073306-6e6e661baf57?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2167&q=8',
    captureComponent: <span className='capture-class'> Image 5th</span>,
  }, {
    src: 'https://images.unsplash.com/photo-1533907650686-70576141c030?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=6000&q=80',
    captureComponent: <span className='capture-class'> some capyture </span>,
  },  {
    src: 'https://images.unsplash.com/photo-1474401915596-3c5adf84ef01?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2250&q=80',
    captureComponent: <span className='capture-class'> some capyture </span>,
  }
];

export default sliderItems;
