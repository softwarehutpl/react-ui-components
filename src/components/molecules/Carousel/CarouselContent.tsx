import React from 'react';
import styled from 'styled-components';

interface CarouselContentProps {
  width?: number;
  transition: number;
  translateCarousel: number;
}

const CarouselContent = styled.div<CarouselContentProps>`
  transform: translateX(-${props => props.translateCarousel}px);
  transition: transform ease-out ${props => props.transition}s;
  height: 100%;
  width: ${props => 800 && props.width}px;
  display: flex;
  overflow: hidden;
`;

export default CarouselContent;
