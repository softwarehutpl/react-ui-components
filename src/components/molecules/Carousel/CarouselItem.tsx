/** @jsx jsx */
import React from 'react'
import { css, jsx } from '@emotion/core'
import styled from '@emotion/styled';

const Slide = ({ content, width, borderRadius, itemsPerSlide, height, margin, carouselWidth}: any) => (
  <div
    css={css`
      height: ${height}px;
      width: ${width ? width + 'px' : carouselWidth/itemsPerSlide + 'px'};
      padding: 0 ${margin/2}px;
    `}
  > <div
     css={css`
      height: 100%;
      border-radius: ${borderRadius}px;
      background-image: url('${content.src}');
      background-size: cover;
      background-repeat: no-repeat;
      background-position: center;
      position: relative;

    `}>
      {content.captureComponent}
  </div>
  </div>
)

const defaultProps = {

}

export default Slide;
