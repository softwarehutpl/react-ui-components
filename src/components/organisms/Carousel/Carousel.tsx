import React from 'react';

export const Carousel: React.StatelessComponent<{}> = () => (
  <div className="">Carousel
  </div>
);
