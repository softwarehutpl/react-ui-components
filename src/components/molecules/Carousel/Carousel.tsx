import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import CarouselContent from './CarouselContent';
import CarouselItem from './CarouselItem';

type Slide = {
  src: string;
  captureComponent?: React.ReactNode;
};

type Arrow = {
  color?: string,
  height?: number,
  backgroundColor? : string,
  width?: 3,
}

interface ButtonProps {
  left?: number;
  right?: number;
  height?: number;
  arrow?: Arrow;
  margin?: number;
  color?: string;
  width?: number;
  backgroundColor?: string;
  top: number;
}

interface CarouselProps {
  slides: Slide[];
  arrow?: Arrow;
  autoslider?: boolean,
  changeTime?: number,
  margin?: number;
  buttonDist?: number;
  itemsPerSlide?: number;
  width?: number;
  height?: number;
  verticalPadding?: number;
  backgroundColor?: string;
}

interface CarouselWrapperProps {
  width?: number;
  verticalPadding?: number;
  backgroundColor?: string;
}

export const Button = styled.button<ButtonProps>`
  position: absolute;
  top: ${({ top }) =>  '150px' && `${top}px`};
  ${({ left, margin }) => margin && left && `left: ${margin + left}px`};
  ${({ right, margin }) => margin && right && `right: ${right + margin}px`};
  transform: translateY(-50%);
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: none;
  background-color: ${({ backgroundColor }) => backgroundColor};
  &:focus {
    outline: none;
  }
  &::before {
    border-style: solid;
    border-width: ${({ width }) => `${width}px ${width}px 0 0`};
    content: '';
    display: inline-block;
    height: ${({ height}) => '40px' && `${height}px`};
    left: 0;
    top: 0.25rem;
    color: ${({ color }) => color};
    position: relative;
    transform: ${({ left }) => left ? 'rotate(-135deg)' : 'rotate(45deg)'};
    vertical-align: top;
    width: ${({ height }) => '40px' && `${height}px`};
  }
`;

export const CarouselWrapper = styled.div<CarouselWrapperProps>`
  position: relative;
  width: ${({ width }) => `${width}px`};
  overflow: hidden;
  background-color: ${({ backgroundColor }) => backgroundColor};
  padding: ${({ verticalPadding }) => `${verticalPadding}px 0`}
`;

/**
 * @function Carousel
 */
const Carousel = (props: CarouselProps) => {
 
  const [state, setState] = useState({
    translateCarousel: 0,
    transition: 0.45,
    activeSlide: 1,
    changeTime: props.changeTime,
  });

  useEffect(() => {
    if (props.autoslider) {
      const timer = setTimeout(() => {
        handleGoToNextSlide();
      }, state.changeTime);
      return () => clearTimeout(timer);
    }
  });

  const updateState = (slideToMove: number) => {
    const { width = 800, itemsPerSlide = 1 } = props;
    setState({ ...state, translateCarousel: width*Math.floor(slideToMove/itemsPerSlide), activeSlide: slideToMove });
  }

  const handleGoToNextSlide = () => {
    const { itemsPerSlide = 1 } = props;
    let slideToMove;
    if (activeSlide >= (props.slides.length - itemsPerSlide)) {
      slideToMove = 0
    } else {
      slideToMove = activeSlide + itemsPerSlide;
    }
    updateState(slideToMove);
  }

  const handleGoToPreviousSlide = () => {
    const { itemsPerSlide = 1 } = props;
    let slideToMove;
    if (activeSlide === 1) {
      slideToMove = props.slides.length;
    } else {
      slideToMove = activeSlide - itemsPerSlide;
    }
    updateState(slideToMove);
  }

  const { translateCarousel, transition, activeSlide } = state;

  return (
    <CarouselWrapper {...props}>
      <CarouselContent
        translateCarousel={translateCarousel}
        transition={transition}
        width={props.width && props.itemsPerSlide && props.width*props.slides.length/props.itemsPerSlide}
      >
        {props.slides.map((slide: Slide, i: number) => (
          <CarouselItem
            key={i}
            content={slide}
            itemsPerSlide={props.itemsPerSlide}
            height={props.height}
            carouselWidth={props.width}
          />
        ))}
      </CarouselContent>
      <Button onClick={() => handleGoToNextSlide()} right={props.buttonDist} {...props.arrow} margin={props.margin} top={props.height ? props.height/2 : 150} />
      <Button onClick={() => handleGoToPreviousSlide()} left={props.buttonDist} {...props.arrow} margin={props.margin} top={props.height ? props.height/2 : 150} />
    </CarouselWrapper>
  )
}

const defaultProps = {
  autoslider: true,
  changeTime: 5000,
  margin: 50,
  buttonDist: 1,
  itemsPerSlide: 1,
  width: 800,
  height: 300,
  verticalPadding: 0,
  item: {
    borderRadius: 30,
    margin: 50,
  },
}

Carousel.defaultProps = defaultProps;

const arrowDeafultProps = {
  color: 'white',
  height: 20,
  backgroundColor: 'transparent',
  width: 3,
  right: 50,
}

Button.defaultProps = arrowDeafultProps;

export default Carousel;
