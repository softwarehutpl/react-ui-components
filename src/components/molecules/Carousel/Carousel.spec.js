import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import sliderItems from '../../../common/mocks/sliderItems';
import Carousel, { CarouselWrapper, Button } from './Carousel';
import CarouselContent from './CarouselContent';
import CarouselItem from './CarouselItem';

describe('Carousel Component', () => {
  it('should render Carousel correctly', () => {
    const tree = create(<Carousel slides={sliderItems} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render CarouselWrapper correctly', () => {
    const tree = create(<CarouselWrapper slides={sliderItems} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Button correctly', () => {
    const tree = create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render CarouselContent correctly', () => {
    const tree = create(<CarouselContent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render CarouselItem correctly', () => {
    const tree = create(<CarouselItem content={sliderItems[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call button onClick method in Button', () => {
    const mockOnChange = jest.fn();
    const component = create(<Button onClick={mockOnChange} />);
    const instance = component.root;
    const button = instance.findByType('button');
    act(() => {
      button.props.onClick();
    });
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });

  it('should render Carousel with default styles', () => {
    const component = create(<Carousel slides={sliderItems} />);
    const carouselWrapper = component.toJSON();
    expect(carouselWrapper).toHaveStyleRule('width', '800px');
    expect(carouselWrapper).toHaveStyleRule('position', 'relative');
    expect(carouselWrapper).toHaveStyleRule('overflow', 'hidden');
  });

    it('should render Carousel with provided styles', () => {
    const component = create(<Carousel slides={sliderItems} width={1000} />);
    const carouselWrapper = component.toJSON();
    expect(carouselWrapper).toHaveStyleRule('width', '1000px');
  });

  it('should render Carausel content', () => {
    const component = create(<CarouselContent translateCarousel={200} transition={0.5} />);
    const modalWrapper = component.toJSON();
    expect(modalWrapper).toHaveStyleRule('height', '100%');
    expect(modalWrapper).toHaveStyleRule('display', 'flex');
    expect(modalWrapper).toHaveStyleRule('transform', 'translateX(-200px)');
    expect(modalWrapper).toHaveStyleRule('transition', 'transform ease-out 0.5s');
  });
});
