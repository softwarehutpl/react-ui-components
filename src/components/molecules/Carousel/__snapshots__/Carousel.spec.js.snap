// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component should render Button correctly 1`] = `
.c0 {
  position: absolute;
  top: undefinedpx;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: none;
  background-color: transparent;
}

.c0:focus {
  outline: none;
}

.c0::before {
  border-style: solid;
  border-width: 3px 3px 0 0;
  content: '';
  display: inline-block;
  height: 20px;
  left: 0;
  top: 0.25rem;
  color: white;
  position: relative;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  vertical-align: top;
  width: 20px;
}

<button
  className="c0"
  color="white"
  height={20}
  width={3}
/>
`;

exports[`Carousel Component should render Carousel correctly 1`] = `
.c1 {
  -webkit-transform: translateX(-0px);
  -ms-transform: translateX(-0px);
  transform: translateX(-0px);
  -webkit-transition: -webkit-transform ease-out 0.45s;
  -webkit-transition: transform ease-out 0.45s;
  transition: transform ease-out 0.45s;
  height: 100%;
  width: 11200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.c2 {
  position: absolute;
  top: 150px;
  right: 51px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: none;
  background-color: transparent;
}

.c2:focus {
  outline: none;
}

.c2::before {
  border-style: solid;
  border-width: 3px 3px 0 0;
  content: '';
  display: inline-block;
  height: 20px;
  left: 0;
  top: 0.25rem;
  color: white;
  position: relative;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  vertical-align: top;
  width: 20px;
}

.c3 {
  position: absolute;
  top: 150px;
  left: 51px;
  right: 100px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: none;
  background-color: transparent;
}

.c3:focus {
  outline: none;
}

.c3::before {
  border-style: solid;
  border-width: 3px 3px 0 0;
  content: '';
  display: inline-block;
  height: 20px;
  left: 0;
  top: 0.25rem;
  color: white;
  position: relative;
  -webkit-transform: rotate(-135deg);
  -ms-transform: rotate(-135deg);
  transform: rotate(-135deg);
  vertical-align: top;
  width: 20px;
}

.c0 {
  position: relative;
  width: 800px;
  overflow: hidden;
  padding: 0px 0;
}

<div
  className="c0"
  height={300}
  width={800}
>
  <div
    className="c1"
    width={11200}
  >
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-pzy0xz-Slide"
      />
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1sz8str-Slide"
      >
        <span
          className="capture-class"
        >
           Image second 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-php6v4-Slide"
      >
        <span
          className="capture-class"
        >
           Image third 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-12otvzt-Slide"
      >
        <span
          className="capture-class"
        >
          Image 4th 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-tftcnt-Slide"
      >
        <span
          className="capture-class"
        >
           Image 5th
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1h2qrbe-Slide"
      >
        <span
          className="capture-class"
        >
           some capyture 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1unxbwy-Slide"
      >
        <span
          className="capture-class"
        >
           some capyture 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-pzy0xz-Slide"
      >
        <span
          className="capture-class"
        >
           Image first 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1sz8str-Slide"
      >
        <span
          className="capture-class"
        >
           Image second 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-php6v4-Slide"
      >
        <span
          className="capture-class"
        >
           Image third 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-12otvzt-Slide"
      >
        <span
          className="capture-class"
        >
          Image 4th 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-tftcnt-Slide"
      >
        <span
          className="capture-class"
        >
           Image 5th
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1h2qrbe-Slide"
      >
        <span
          className="capture-class"
        >
           some capyture 
        </span>
      </div>
    </div>
    <div
      className="css-12155ub-Slide"
    >
       
      <div
        className="css-1unxbwy-Slide"
      >
        <span
          className="capture-class"
        >
           some capyture 
        </span>
      </div>
    </div>
  </div>
  <button
    className="c2"
    color="white"
    height={20}
    onClick={[Function]}
    width={3}
  />
  <button
    className="c3"
    color="white"
    height={20}
    onClick={[Function]}
    width={3}
  />
</div>
`;

exports[`Carousel Component should render CarouselContent correctly 1`] = `
.c0 {
  -webkit-transform: translateX(-px);
  -ms-transform: translateX(-px);
  transform: translateX(-px);
  -webkit-transition: -webkit-transform ease-out s;
  -webkit-transition: transform ease-out s;
  transition: transform ease-out s;
  height: 100%;
  width: px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

<div
  className="c0"
/>
`;

exports[`Carousel Component should render CarouselItem correctly 1`] = `
<div
  className="css-bd7xg4-Slide"
>
   
  <div
    className="css-pzy0xz-Slide"
  />
</div>
`;

exports[`Carousel Component should render CarouselWrapper correctly 1`] = `
.c0 {
  position: relative;
  width: undefinedpx;
  overflow: hidden;
  padding: undefinedpx 0;
}

<div
  className="c0"
/>
`;
